docs/atomic.rst

```diff
@@ -0,0 +1,96 @@
+.. _atomic:
+
+Atomic Operations
+=================
+
+Sometimes we want to access a common variable in different GPU threads. For example, we want to sum up the values in tensor ``x``:
+::
+    @ti.kernel
+    def sum():
+        for i in x:
+            result[None] = result[None] + x[i]
+
+This will be translated to (in GPU pseudo-code):
+::
+    read from ``result`` to register 1
+    read from ``x[i]`` to register 2
+    add the value of reg1 and reg2, save result to reg3
+    write back the value in register 3 to ``result``
+
+This code is executed by many parallel GPU threads for different ``i``.
```

Comment:
A high level comment - I have a feeling that most users are aware of what atomics are, and what problem atomics can solve. Maybe just mention the fact that Taichi supports atomic operations on both integers and floats?

Reply:
I thought we taichi is a repo for getting CG newcomers?
> ...on both integers and floats...
OK, let's also add a table for supported atomic operation of different backends.

Reply:
I think here we should refer readers to https://en.wikipedia.org/wiki/Fetch-and-add for an explanation of `atomic_add` and simply say `x[i] += 1` is atomic but `x[i] = x[i] + 1` is not. Explaining what atomic operations mean should probably not be part of taichi's documentation. (But feel free to write a blog post with these details! People will appreciate that :-)